mentat/code_context.py

```diff
@@ -242,7 +247,10 @@ def get_code_message(self, model: str, code_file_manager: "CodeFileManager") ->
             file_lines = code_file_manager.file_lines[rel_path]
             for i, line in enumerate(file_lines, start=1):
                 if file.contains_line(i):
-                    file_message.append(f"{i}:{line}")
+                    if parser.provide_line_numbers():
```

Comment:
It seems a little finicky to have to pass this one bit of information down from the parser. It's fine for this PR and maybe there won't be a need to pass more information from the parser to the code formater. But if we do end up doing it all the time we could consider giving the parser the responsibility of printing out the file.

Reply:
I agree it is a bit weird haha. I do like the idea of the parser printing out the file if we get more parser specific printing instructions.